service/secret/secret.go

```diff
@@ -0,0 +1,9 @@
+package secret
+
+import "context"
+
+// Store describes a type that can securely obtain secrets for services.
+type Store interface {
+	GetSecretsForTarget(name string) (map[string]string, error)
+	Renew(ctx context.Context) error
```

Comment:
I still dislike this. Should move it inside the actual Vault secret store constructor.